src/core/server/metrics/metrics_service.test.ts

```diff
@@ -118,6 +121,168 @@ describe('MetricsService', () => {
       expect(await nextEmission()).toEqual({ metric: 'first' });
       expect(await nextEmission()).toEqual({ metric: 'second' });
     });
+
+    it('logs the metrics at every interval', async () => {
+      const firstMetrics = {
+        process: {
+          memory: { heap: { used_in_bytes: 100 } },
+          uptime_in_millis: 1500,
+          event_loop_delay: 50,
+        },
+        os: {
+          load: {
+            '1m': 10,
+            '5m': 20,
+            '15m': 30,
+          },
+        },
+      };
+      const secondMetrics = {
+        process: {
+          memory: { heap: { used_in_bytes: 200 } },
+          uptime_in_millis: 3000,
+          event_loop_delay: 100,
+        },
+        os: {
+          load: {
+            '1m': 20,
+            '5m': 30,
+            '15m': 40,
+          },
+        },
+      };
+
+      const opsLogger = logger.get('metrics', 'ops');
+
+      mockOpsCollector.collect
+        .mockResolvedValueOnce(firstMetrics)
+        .mockResolvedValueOnce(secondMetrics);
+      await metricsService.setup({ http: httpMock });
+      const { getOpsMetrics$ } = await metricsService.start();
+
+      const nextEmission = async () => {
+        jest.advanceTimersByTime(testInterval);
+        const emission = await getOpsMetrics$().pipe(take(1)).toPromise();
+        await new Promise((resolve) => process.nextTick(resolve));
+        return emission;
+      };
+      expect(loggingSystemMock.collect(opsLogger).debug[1]).toMatchInlineSnapshot(`
+        Array [
```

Comment:
optional: the output shape is already covered by `get_ops_metrics_log` tests.

Reply:
We're only interested in checking that the values aren't the same and that what is logged out is the most recent data emitted and not specifically looking at the output shape.

Reply:
👍